src/promptflow/promptflow/contracts/run_info.py

```diff
@@ -228,3 +228,22 @@ def deserialize(data: dict) -> "FlowRunInfo":
             upload_metrics=data.get("upload_metrics", False),
         )
         return flow_run_info
+
+    @staticmethod
+    def create_with_error(start_time, inputs, index, run_id, error):
+        return FlowRunInfo(
+            run_id=run_id,
+            status=Status.Failed,
+            error=error,
```

Comment:
Here `error` has to be a dict right? Do we need to check it first?

Reply:
I think we don't need to check it.

- For expected error, C# executor will return 200+line result
- For unexpected error, batch engine will wrap it into UnexpectedError and turn it to error dict.